src/plugins/gui/providers/revolutProvider.ts

```diff
@@ -0,0 +1,278 @@
+          amount: params.exchangeAmount,
+          crypto: revolutCrypto.id,
+          payment: 'revolut', // Only revolut is supported at the moment
+          region: params.regionCode.countryCode
+        })
+
+        const cryptoAmount = quoteData.crypto.amount.toString()
+        const fiatAmount = params.exchangeAmount
+
+        return {
+          providerId,
+          partnerIcon,
+          pluginDisplayName,
+          displayCurrencyCode: params.displayCurrencyCode,
+          cryptoAmount,
+          isEstimate: false,
+          fiatCurrencyCode: params.fiatCurrencyCode,
+          fiatAmount,
+          direction: params.direction,
+          expirationDate: undefined,
+          regionCode: params.regionCode,
+          paymentTypes: params.paymentTypes,
+
+          approveQuote: async (approveParams: FiatProviderApproveQuoteParams) => {
+            const { showUi, coreWallet } = approveParams
+            const walletAddresses = await coreWallet.getAddresses({ tokenId: params.tokenId })
+            const walletAddress = walletAddresses[0]?.publicAddress
+
+            if (walletAddress == null) {
+              throw new Error('No wallet address found')
+            }
+
+            const successReturnURL = encodeURIComponent('https://return.edge.app/fiatprovider/buy/revolut?transactionStatus=success')
+
+            const orderId = await io.makeUuid()
+
+            const { ramp_redirect_url: redirectUrl } = await fetchRevolutRedirectUrl({
+              fiat: revolutFiat.currency,
+              amount: parseFloat(fiatAmount),
+              crypto: quoteData.crypto.currencyId,
+              payment: 'revolut',
+              region: params.regionCode.countryCode,
+              wallet: walletAddress,
+              partnerRedirectUrl: successReturnURL,
+              orderId
+            })
+
+            console.log('!@! redirectUrl', redirectUrl)
+
+            await showUi.openExternalWebView({
+              url: redirectUrl,
+              providerId,
+              async deeplinkHandler(link) {
+                if (link.direction === 'sell') {
+                  throw new Error('Unexpected sell return link. Sell not supported by Revolut.')
+                }
+                const { transactionStatus } = link.query
+                if (transactionStatus === 'success') {
+                  await showUi.trackConversion('Buy_Success', {
+                    conversionValues: {
+                      conversionType: 'buy',
+                      sourceFiatCurrencyCode: params.fiatCurrencyCode,
+                      sourceFiatAmount: fiatAmount,
+                      destAmount: new CryptoAmount({
+                        currencyConfig: coreWallet.currencyConfig,
+                        currencyCode: params.displayCurrencyCode,
+                        exchangeAmount: cryptoAmount
+                      }),
+                      fiatProviderId: providerId,
+                      orderId
```

Comment:
Please absolutely triple check that the `orderId` is exactly what gets returned from their reporting API as these need to match to give affiliates credit for conversions.